frontend_tests/casper_tests/05-subscriptions.js

```diff
@@ -131,25 +131,43 @@ casper.then(function () {
 });
 casper.then(function () {
     casper.click('#create_stream_button');
-    casper.fill('form#stream_creation_form', {stream_name: '  '});
-    casper.click('form#stream_creation_form button.btn.btn-primary');
+    casper.waitUntilVisible("form#stream_creation_form", function () {
+        casper.fill('form#stream_creation_form', {stream_name: '  '});
+        casper.click('form#stream_creation_form button.btn.btn-primary');
+    });
 });
 casper.then(function () {
     casper.waitForSelectorText('#stream_name_error', 'A stream needs to have a name', function () {
         casper.test.assertTextExists('A stream needs to have a name', "Can't create a stream with an empty name");
-        casper.click('form#stream_creation_form button.btn.btn-default');
-        casper.fill('form#add_new_subscription', {stream_name: '  '});
-        casper.click('#create_stream_button');
-        casper.fill('form#stream_creation_form', {stream_name: 'Waseemio'});
-        casper.click('form#stream_creation_form button.btn.btn-primary');
+        casper.waitUntilVisible("form#stream_creation_form", function () {
```

Comment:
I think you can remove this block. We have already tested the empty stream name above.

Reply:
@umairwaheed Without `casper.fill('form#add_new_subscription', {stream_name: '  '});` The tests are failing. I fail to understand what's going wrong.